nikola/post.py

```diff
@@ -1023,6 +1031,28 @@ def _get_metadata_from_file(meta_data):
     if not meta_data[0]:
         meta_data = meta_data[1:]
 
+    # If 1st line is '---', then it's YAML metadata
```

Comment:
It works just fine, because we are removing metadata before feeding it to docutils.

It does mean you can't use docutils rst2html with a post anymore.